packages/terra-search-field/src/SearchField.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import IconSearch from 'terra-icon/lib/icon/IconSearch';
+import classNames from 'classnames';
+import './SearchField.scss';
+
+const propTypes = {
+  /**
+   * Placeholder text to show while the search field is empty.
+   */
+  placeholder: PropTypes.string,
+
+  /**
+   * The minimum number of characters to perform a search.
+   */
+  minimumSearchTextLength: PropTypes.number,
+
+  /**
+   * How long the component should wait after input before performing an automatic search.
+   */
+  searchDelay: PropTypes.number,
+
+  /**
+   * A callback to perform search. Sends parameter {String} searchText.
+   */
+  onSearch: PropTypes.func,
+};
```

Comment:
I would get a few other opinions on this before implementing anything, but we might need to supply an inputAttributes prop to spread onto the input field. The date picker does something similar: https://github.com/cerner/terra-core/blob/master/packages/terra-date-picker/src/DateInput.jsx#L13

There's a lot of use cases where consumers want/need to pass down a name or id to the input field. (or their own custom events, onBlur, onFocus etc.. )